ui/app/src/Utils/FormValidation.elm

```diff
@@ -0,0 +1,62 @@
+module Utils.FormValidation
+    exposing
+        ( ValidationResult(Valid, NotValid)
+        , ValidatedString
+        , validateDate
+        , areValidResults
+            , validatedMatcherToMatcher
+        , ValidatedMatcher
+        )
+
+import Utils.Date exposing (timeFromString)
+import Silences.Types exposing (Silence)
+import Utils.Types exposing (Matcher)
+
+
+type ValidationResult
+    = Valid
+    | NotValid String
+
+
+type alias ValidatedString =
+    { value : String
+    , result : ValidationResult
+    }
```

Comment:
This is piggy-backing on @w0rm's comment: I think we can get rid of the type alias, and squash all information into the union type:

```
type ValidationResult
    = Valid String
    | NotValid String
```

That way when you do `case validation of`, you can just access the value directly.